SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,485 @@
+				const Worker_EntityId EntityId = Delta.EntityId;
+				Handler(EntityId);
+			}
+		}
+	}
+};
+
+FDistributedStartupActorSkeletonEntityCreator::FDistributedStartupActorSkeletonEntityCreator(USpatialNetDriver& InNetDriver)
+	: NetDriver(&InNetDriver)
+	, ManifestSubView(&NetDriver->Connection->GetCoordinator().CreateSubView(
+		  FSkeletonEntityManifest::ComponentId, FSubView::NoFilter,
+		  { NetDriver->Connection->GetCoordinator().CreateComponentChangedRefreshCallback(FSkeletonEntityManifest::ComponentId) }))
+{
+}
+
+void FDistributedStartupActorSkeletonEntityCreator::CreateSkeletonEntitiesForLevel(UWorld& World)
+{
+	ensure(Stage == EStage::Initial);
+
+	UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Creating skeleton entities for world %s"), *World.GetName())
+
+	Stage = EStage::CreatingEntities;
+
+	TSet<Worker_EntityId_Key> SkeletonEntityIds;
+
+	for (AActor* Actor : TActorRange<AActor>(&World))
+	{
+		if (!IsValid(Actor))
+		{
+			continue;
+		}
+
+		if (!(Actor->IsNameStableForNetworking() && Actor->GetIsReplicated()))
+		{
+			continue;
+		}
+
+		const Worker_EntityId SkeletonEntityId = CreateSkeletonEntityForActor(*Actor);
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("EntityId: %lld Creating skeleton entity for %s"), SkeletonEntityId,
+			   *Actor->GetName());
+
+		RemainingSkeletonEntities.Emplace(SkeletonEntityId);
+		SkeletonEntityIds.Emplace(SkeletonEntityId);
+	}
+
+	if (RemainingSkeletonEntities.Num() == 0)
+	{
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("No skeleton entities must be created for %s"), *World.GetName())
+
+		Stage = EStage::DelegatingEntities;
+	}
+}
+
+Worker_EntityId FDistributedStartupActorSkeletonEntityCreator::CreateSkeletonEntityForActor(AActor& Actor)
+{
+	const Worker_EntityId ActorEntityId = NetDriver->PackageMap->AllocateEntityId();
+
+	EntityFactory Factory(NetDriver, NetDriver->PackageMap, NetDriver->ClassInfoManager, NetDriver->RPCService.Get());
+
+	TArray<FWorkerComponentData> EntityComponents = Factory.CreateSkeletonEntityComponents(&Actor);
+
+	// LB components also contain authority delegation, giving this worker ServerAuth.
+	Factory.WriteLBComponents(EntityComponents, &Actor);
+
+	// RPC components.
+	auto RPCService = NetDriver->RPCService.Get();
+
+	checkf(RPCService != nullptr, TEXT("Attempting to create an entity with a null RPCService."));
+	EntityComponents.Append(RPCService->GetRPCComponentsOnEntityCreation(ActorEntityId));
+	EntityComponents.Append(NetDriver->RPCs->GetRPCComponentsOnEntityCreation(ActorEntityId));
+
+	// Skeleton entity markers		.
+	EntityComponents.Emplace(ComponentFactory::CreateEmptyComponentData(SpatialConstants::FLESHOUT_QUERY_TAG_COMPONENT_ID));
+	EntityComponents.Emplace(ComponentFactory::CreateEmptyComponentData(SpatialConstants::FLESHOUT_REQUIRED_TAG_COMPONENT_ID));
+
+	EntityComponents.Add(NetDriver->InterestFactory->CreateInterestData(
+		&Actor, NetDriver->ClassInfoManager->GetOrCreateClassInfoByObject(&Actor), ActorEntityId));
```

Comment:
Consider custom interest for the populating server only

Reply:
Would be bad to do so as non-auth workers need to see skeleton entities for actors that are in their interest.

Reply:
We'd better use `AddServerSelfInterest` here